### PR TITLE
[ODS-5212]Migration Support for ODS / API 5.2

### PR DESCRIPTION
**Pull request by semalaiappan**

### CODE REVIEW

File: EdFi.Ods.Utilities.Migration/Scripts/MsSql/02Upgrade/v51_to_v52_RecordOwnership/01 Structure/1000-AddOwnershipColumnsForNewTablesV52.sql [hunk 1]
```diff
@@ -0,0 +1,8 @@
+-- SPDX-License-Identifier: Apache-2.0
+-- Licensed to the Ed-Fi Alliance under one or more agreements.
+-- The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+-- See the LICENSE and NOTICES files in the project root for more information.
+
+ALTER TABLE [edfi].[AssessmentScoreRangeLearningStandard] ADD [CreatedByOwnershipTokenId] SMALLINT NULL;
+ALTER TABLE [edfi].[StudentDisciplineIncidentBehaviorAssociation] ADD [CreatedByOwnershipTokenId] SMALLINT NULL;
+ALTER TABLE [edfi].[StudentDisciplineIncidentNonOffenderAssociation] ADD [CreatedByOwnershipTokenId] SMALLINT NULL;
```
Comment (gmcelhanon) [inline EdFi.Ods.Utilities.Migration/Scripts/MsSql/02Upgrade/v51_to_v52_RecordOwnership/01 Structure/1000-AddOwnershipColumnsForNewTablesV52.sql:8]: Comment is actually for the rename operation below. Why are we renaming several files to v53 in a branch for API 5.2?

File: EdFi.Ods.Utilities.Migration/Scripts/PgSql/02Upgrade/v51_to_v52/01 Structure/1040-AddExtendedPropertiesForV52.sql [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+IS 'A unique number or alphanumeric code assigned to the score range associated with one or more learning standards.';
+
+COMMENT ON TABLE edfi.AssessmentScoreRangeLearningStandardLearningStandard
+IS 'LearningStandard associated with the score range.';
+
+COMMENT ON TABLE edfi.DisciplineActionStudentDisciplineIncidentBehaviorAssociation
+IS 'A reference to the behavior(s) by the student that led or contributed to this specific action.';
+
+COMMENT ON COLUMN edfi.DisciplineActionStudentDisciplineIncidentBehaviorAssociation.BehaviorDescriptorId
+IS 'Describes behavior by catery.';
```
Comment (gmcelhanon) [inline EdFi.Ods.Utilities.Migration/Scripts/PgSql/02Upgrade/v51_to_v52/01 Structure/1040-AddExtendedPropertiesForV52.sql]: Typo: `catery` should be `category`.

File: EdFi.Ods.Utilities.Migration/Scripts/PgSql/02Upgrade/v51_to_v52/01 Structure/1040-AddExtendedPropertiesForV52.sql [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+COMMENT ON COLUMN edfi.StudentDisciplineIncidentBehaviorAssociation.SchoolId IS 'The identifier assigned to a school.' ;
+
+
+COMMENT ON COLUMN edfi.StudentDisciplineIncidentBehaviorAssociation.StudentUSI IS 'A unique alphanumeric code assigned to a student.';
+
+
+COMMENT ON TABLE edfi.StudentDisciplineIncidentBehaviorAssociation IS 'This association describes the behavior of students involved in a discipline incident.';
+
+
+COMMENT ON COLUMN edfi.StudentDisciplineIncidentBehaviorAssociationDisciplineIn_ae6a21.BehaviorDescriptorId IS 'Describes behavior by catery.';
```
Comment (gmcelhanon) [inline EdFi.Ods.Utilities.Migration/Scripts/PgSql/02Upgrade/v51_to_v52/01 Structure/1040-AddExtendedPropertiesForV52.sql]: Typo: `catery` should be `category`.

File: EdFi.Ods.Utilities.Migration/Scripts/PgSql/02Upgrade/v51_to_v52/01 Structure/1040-AddExtendedPropertiesForV52.sql [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+IS 'The original peoples or cultures with which the individual identifies.';
+
+
+COMMENT ON COLUMN edfi.StaffAncestryEthnicOrigin.StaffUSI IS 'A unique alphanumeric code assigned to a staff.' ;
+
+
+COMMENT ON TABLE edfi.StaffAncestryEthnicOrigin IS 'The original peoples or cultures with which the individual identifies.' ;
+
+
+COMMENT ON COLUMN edfi.StudentDisciplineIncidentBehaviorAssociation.BehaviorDescriptorId IS 'Describes behavior by catery.';
```
Comment (gmcelhanon) [inline EdFi.Ods.Utilities.Migration/Scripts/PgSql/02Upgrade/v51_to_v52/01 Structure/1040-AddExtendedPropertiesForV52.sql]: Typo: `catery` should be `category`.

File: EdFi.Ods.Utilities.Migration/Scripts/PgSql/02Upgrade/v51_to_v52/01 Structure/1010-AddNewTablesForV52.sql [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+    Namespace                             VARCHAR (255)   NOT NULL,
+    ScoreRangeId                          VARCHAR (60)    NOT NULL,
+    AssessmentReportingMethodDescriptorId INT              NULL,
+    MinimumScore                          VARCHAR (35)    NOT NULL,
+    MaximumScore                          VARCHAR (35)    NOT NULL,
+    IdentificationCode                    VARCHAR (60)    NULL,
+    Discriminator                         VARCHAR (128)   NULL,
+    CreateDate                            TIMESTAMP   DEFAULT current_timestamp NOT NULL ,
+    LastModifiedDate                      TIMESTAMP   DEFAULT current_timestamp NOT NULL ,
+    Id                                    UUID  DEFAULT gen_random_uuid() NOT NULL,
```
Comment (gmcelhanon) [inline EdFi.Ods.Utilities.Migration/Scripts/PgSql/02Upgrade/v51_to_v52/01 Structure/1010-AddNewTablesForV52.sql]: In ODS for v5.2 from the package, this line seems like it should end as `UUID  DEFAULT public.gen_random_uuid() NOT NULL,`

File: EdFi.Ods.Utilities.Migration/Scripts/PgSql/02Upgrade/v51_to_v52/01 Structure/1010-AddNewTablesForV52.sql [hunk 6]
```diff
@@ -0,0 +1,245 @@
...
+CREATE TABLE edfi.StudentDisciplineIncidentBehaviorAssociation (
+    BehaviorDescriptorId        INT              NOT NULL,
+    IncidentIdentifier          VARCHAR (20)    NOT NULL,
+    SchoolId                    INT              NOT NULL,
+    StudentUSI                  INT              NOT NULL,
+    BehaviorDetailedDescription VARCHAR (1024)  NULL,
+    Discriminator               VARCHAR (128)   NULL,
+    CreateDate                  TIMESTAMP    DEFAULT current_timestamp NOT NULL,
+    LastModifiedDate            TIMESTAMP    DEFAULT current_timestamp NOT NULL,
+    Id                          UUID DEFAULT gen_random_uuid() NOT NULL,
```
Comment (gmcelhanon) [inline EdFi.Ods.Utilities.Migration/Scripts/PgSql/02Upgrade/v51_to_v52/01 Structure/1010-AddNewTablesForV52.sql]: In ODS for v5.2 from the package, this line seems like it should end as `UUID  DEFAULT public.gen_random_uuid() NOT NULL,`

File: EdFi.Ods.Utilities.Migration/Scripts/PgSql/02Upgrade/v51_to_v52/01 Structure/1010-AddNewTablesForV52.sql [hunk 7]
```diff
@@ -0,0 +1,245 @@
...
+
+
+CREATE TABLE edfi.StudentDisciplineIncidentNonOffenderAssociation (
+    IncidentIdentifier VARCHAR (20)    NOT NULL,
+    SchoolId           INT              NOT NULL,
+    StudentUSI         INT              NOT NULL,
+    Discriminator      VARCHAR (128)   NULL,
+    CreateDate         TIMESTAMP    DEFAULT current_timestamp NOT NULL,
+    LastModifiedDate   TIMESTAMP    DEFAULT current_timestamp NOT NULL,
+    Id                 UUID  DEFAULT gen_random_uuid() NOT NULL,
```
Comment (gmcelhanon) [inline EdFi.Ods.Utilities.Migration/Scripts/PgSql/02Upgrade/v51_to_v52/01 Structure/1010-AddNewTablesForV52.sql]: In ODS for v5.2 from the package, this line seems like it should end as `UUID  DEFAULT public.gen_random_uuid() NOT NULL,`

**Review by gmcelhanon — CHANGES_REQUESTED** [review]
Both migrations worked without error. There were some minor (probably whitespace) differences on an extended property (on `edfi.StudentDisciplineIncidentAssociation` table in SQL Server and PostgreSQL). Additionally, there were a few occurrences of a very minor discrepancy on the definition of the `Id` column on some tables, and there was a typo in the comment for the `BehaviorDescriptorId` column in 3 places.

File: EdFi.Ods.Utilities.Migration/Scripts/PgSql/02Upgrade/v51_to_v52/01 Structure/1040-AddExtendedPropertiesForV52.sql (hunk 2 above)
Comment (semalaiappan) [reply EdFi.Ods.Utilities.Migration/Scripts/PgSql/02Upgrade/v51_to_v52/01 Structure/1040-AddExtendedPropertiesForV52.sql]: fixed it

File: EdFi.Ods.Utilities.Migration/Scripts/PgSql/02Upgrade/v51_to_v52/01 Structure/1040-AddExtendedPropertiesForV52.sql (hunk 3 above)
Comment (semalaiappan) [reply EdFi.Ods.Utilities.Migration/Scripts/PgSql/02Upgrade/v51_to_v52/01 Structure/1040-AddExtendedPropertiesForV52.sql]: fixed it

File: EdFi.Ods.Utilities.Migration/Scripts/PgSql/02Upgrade/v51_to_v52/01 Structure/1040-AddExtendedPropertiesForV52.sql (hunk 4 above)
Comment (semalaiappan) [reply EdFi.Ods.Utilities.Migration/Scripts/PgSql/02Upgrade/v51_to_v52/01 Structure/1040-AddExtendedPropertiesForV52.sql]: fixed it

File: EdFi.Ods.Utilities.Migration/Scripts/PgSql/02Upgrade/v51_to_v52/01 Structure/1010-AddNewTablesForV52.sql (hunk 5 above)
Comment (semalaiappan) [reply EdFi.Ods.Utilities.Migration/Scripts/PgSql/02Upgrade/v51_to_v52/01 Structure/1010-AddNewTablesForV52.sql]: fixed it

File: EdFi.Ods.Utilities.Migration/Scripts/PgSql/02Upgrade/v51_to_v52/01 Structure/1010-AddNewTablesForV52.sql (hunk 6 above)
Comment (semalaiappan) [reply EdFi.Ods.Utilities.Migration/Scripts/PgSql/02Upgrade/v51_to_v52/01 Structure/1010-AddNewTablesForV52.sql]: fixed it

File: EdFi.Ods.Utilities.Migration/Scripts/PgSql/02Upgrade/v51_to_v52/01 Structure/1010-AddNewTablesForV52.sql (hunk 7 above)
Comment (semalaiappan) [reply EdFi.Ods.Utilities.Migration/Scripts/PgSql/02Upgrade/v51_to_v52/01 Structure/1010-AddNewTablesForV52.sql]: fixed it

**Review by gmcelhanon — APPROVED** [review]

**Pull request merged**